controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook.go

```diff
@@ -64,6 +65,40 @@ func (in *KubeadmControlPlane) Default() {
 	if !strings.HasPrefix(in.Spec.Version, "v") {
 		in.Spec.Version = "v" + in.Spec.Version
 	}
+
+	ios1 := intstr.FromInt(1)
+	ios0 := intstr.FromInt(0)
+	// Enforce RollingUpdate strategy and default MaxSurge and MaxUnavailable if not set.
+	if in.Spec.RolloutStrategy != nil {
+		if in.Spec.RolloutStrategy.Type != RollingUpdateStrategyType {
+			in.Spec.RolloutStrategy.Type = RollingUpdateStrategyType
+		}
```

Comment:
It seems odd to try and override the type here if a user explicitly set it, I'm wondering if it might be better to only set it here if it is not set, and then during validation ensure that the type is `RollingUpdateStrategyType`?
```suggestion
		if in.Spec.RolloutStrategy.Type == RolloutStrategyType("") {
			in.Spec.RolloutStrategy.Type = RollingUpdateStrategyType
		}
```